src/study.js

```diff
@@ -1,137 +1,114 @@
+    }, options);
+
+    if (!this.store) throw new Error('You must supply a store!');
+
+    this.previousAssignments = {};
+    try {
+      const data = this.store.get(this.storageKey);
+      if (data && data[0] === '{') {
+        this.previousAssignments = JSON.parse(data);
+      }
+    } catch (_) {
+      // ignore
     }
 
-    this.userBuckets = {};
     this.userAssignments = {};
+    this.persistedUserAssignments = {};
     this.providedTests = [];
-    this.classList = {};
-
-    const userBuckets = this.store.get(storageKey);
-    if (userBuckets) {
-      try {
-        this.userBuckets = JSON.parse(userBuckets);
-      } catch (e) {
-        this.userBuckets = {};
-      }
-    }
   }
 
-  /**
-   * Define a single or multiple tests
-   * @param {Object} data An object or an array of objects containing the test
-   * info such as name and buckets
-   */
-  define(data) {
-    let normalizedData = data;
-    if (!Array.isArray(data)) { normalizedData = [data]; }
+  define(tests) {
+    let normalizedData = tests;
+    if (!Array.isArray(tests)) normalizedData = [tests];
 
     normalizedData.forEach((test) => {
-      if (!test.name) { throw new Error('Tests must have a name'); }
-      if (!test.buckets) { throw new Error('Tests must have buckets'); }
-      if (!Object.keys(test.buckets)) { throw new Error('Tests must have buckets'); }
-      this.providedTests = this.providedTests.concat(test);
+      if (!test.name) throw new Error('Tests must have a name');
+      if (!test.buckets) throw new Error('Tests must have buckets');
+      if (!Object.keys(test.buckets)) throw new Error('Tests must have buckets');
+      this.providedTests.push(test);
     });
   }
 
-  /**
-   * Return all defined tests
-   * @return {Array} Formatted/provided tests
-   */
   definitions() {
-    return response(this.providedTests);
+    return this.providedTests;
   }
 
-  /**
-   * Executes the bucketing of all provided tests. Will bucket a user if the
-   * query string is set, then falls back to restoring a persisted bucket, then
-   * falls back to determining a bucket for the user.
-   * @type {Function}
-   * @param {Boolean} isDry If true, will not persist test buckets
-   */
-  assign(testName, bucketName) {
-    const applyClasses = () => {
-      Object.keys(this.classList).forEach((name) => {
-        const classesToRemove = document.body.className.split(/\s+/g)
-          .map(x => x.trim())
-          .filter(Boolean)
-          .filter(x => x.indexOf(`${name}--`) === 0);
-        classesToRemove.forEach(className => document.body.classList.remove(className));
-        if (this.classList[name]) document.body.classList.add(`${name}--${this.classList[name]}`);
-      });
-      document.removeEventListener('DOMContentLoaded', applyClasses);
-    };
+  removeClasses(testName, exceptClassName) {
+    const { root } = this;
+    if (!root) return;
+
+
+    const currentClassNames = root.className.split(/\s+/g)
```

Comment:
Could we use [classList](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) here instead?